lib/chai/interface/assert.js

```diff
@@ -1232,66 +1232,78 @@ module.exports = function (chai, util) {
-   * `constructor`, or alternately that it will throw an error with message
-   * matching `regexp`.
+   * Asserts that `fn` will throw an error that is an instance of `errorLike`
+   * (if `errorLike` is an `Error` constructor) or if it is the same instance as `errorLike`
+   * (if `errorLike` is an `Error` instance). This also asserts that the error thrown
+   * will have a message matching `errMsgMatcher`.
    *
    *     assert.throws(fn, 'function throws a reference error');
    *     assert.throws(fn, /function throws a reference error/);
    *     assert.throws(fn, ReferenceError);
-   *     assert.throws(fn, ReferenceError, 'function throws a reference error');
-   *     assert.throws(fn, ReferenceError, /function throws a reference error/);
+   *     assert.throws(fn, new ReferenceError());
+   *     assert.throws(fn, ReferenceError, 'Error thrown must be a ReferenceError and have this msg');
+   *     assert.throws(fn, new ReferenceError(), 'Error thrown must be the same ReferenceError instance and have this msg');
+   *     assert.throws(fn, ReferenceError, /Error thrown must be a ReferenceError and match this/);
+   *     assert.throws(fn, new ReferenceError(), /Error thrown must be the same ReferenceError instance and match this/);
    *
    * @name throws
    * @alias throw
    * @alias Throw
-   * @param {Function} function
-   * @param {ErrorConstructor} constructor
-   * @param {RegExp} regexp
+   * @param {Function} fn
+   * @param {ErrorConstructor/Error} errorLike
+   * @param {RegExp/String} errMsgMatcher
    * @param {String} message
    * @see https://developer.mozilla.org/en/JavaScript/Reference/Global_Objects/Error#Error_types
    * @namespace Assert
    * @api public
    */
 
-  assert.throws = function (fn, errt, errs, msg) {
-    if ('string' === typeof errt || errt instanceof RegExp) {
-      errs = errt;
-      errt = null;
+  assert.throws = function (fn, errorLike, errMsgMatcher, msg) {
+    if ('string' === typeof errorLike || errorLike instanceof RegExp) {
+      errMsgMatcher = errorLike;
+      errorLike = null;
     }
 
-    var assertErr = new Assertion(fn, msg).to.throw(errt, errs);
+    var assertErr = new Assertion(fn, msg).to.throw(errorLike, errMsgMatcher);
     return flag(assertErr, 'object');
   };
 
   /**
-   * ### .doesNotThrow(function, [constructor/regexp], [message])
+   * ### .doesNotThrow(fn, [errorLike/string/regexp], [string/regexp], [message])
    *
-   * Asserts that `function` will _not_ throw an error that is an instance of
-   * `constructor`, or alternately that it will not throw an error with message
-   * matching `regexp`.
+   * Asserts that `fn` will _not_ throw an error that is an instance of `errorLike`
+   * (if `errorLike` is an `Error` constructor) or if it is the same instance as `errorLike`
+   * (if `errorLike` is an `Error` instance). This also asserts that the error thrown
+   * will _not_ have a message matching `errMsgMatcher`.
```

Comment:
Same thing. Consider reordering. The readability is a little worse on this one than the previous one.

```js
If `errorLike` is an `Error` constructor, assert that `fn` won't throw an error that is an instance of `errorLike`.
If `errorLike` is an `Error` instance, assert that `fn` won't throw the same instance as `errorLike`.
If `errMsgMatcher` is provided, also assert that the error thrown won't have a message matching `errMsgMatcher`.
```